### PR TITLE
First pass at ND routing in TT.

### PR DESCRIPTION
**Pull request by acompa**

### CODE REVIEW

**Comment by r0ymanesco** [comment]
I updated the [API](https://www.notion.so/notdiamond/Backend-APIs-0b6f7fc05e374913862b6b9182c20e27) to include `llm_providers` so that you can specify the subset off models you want to use within what Tomas defined. Also renamed `prompt` to `messages` and now you can just submit a list of messages as you would normally, saw that in your PR

File: utils/llm.py [hunk 1]
```diff
@@ -144,7 +144,8 @@ def route_llm_prompt(text_chunk, instructions, source_langage, target_language):
     body = {
         "source_language": source_langage,
         "target_language": target_language,
-        "prompt": [{"role": "system", "content": instructions}, {"role": "user", "content": text_chunk}]
+        "messages": [{"role": "system", "content": instructions}, {"role": "user", "content": text_chunk}],
+        "llm_providers": llm_providers
     }
     response = requests.post(
```
Comment (r0ymanesco) [inline utils/llm.py:149]: The response unpacking is incorrect. You need to do `select_model = response.json().get("provider").get("model")`

**Review by r0ymanesco — COMMENTED** [review]
Just one minor bug otherwise LGTM

**Review by r0ymanesco — APPROVED** [review]

File: utils/llm.py [hunk 2]
```diff
@@ -124,7 +127,67 @@ def find_json(text):
...
     else:
-        return [] 
+        return []
+
+
+def route_llm_prompt(text_chunk, instructions, llm_providers, source_langage, target_language):
+    """
+    Route the given prompt to the appropriate LLM model using Not Diamond.
+    """
+    nd_url = "https://not-diamond-server.onrender.com"
```
Comment (r0ymanesco) [inline utils/llm.py]: This url is wrong. Should be "https://not-diamond-server.onrender.com/v2/TT/translate"

Comment (acompa) [reply utils/llm.py]: Updated, thanks for catching.

**Pull request merged**